src/Sarif.Multitool.FunctionalTests/Rules/ValidationSkimmerTestsBase.cs

```diff
@@ -87,15 +88,31 @@ protected void Verify(string testFileName, bool disablePrereleaseCompatibilityTr
         // Verify that those detected result locations match the expected locations.
         private void Verify(Run run, ExpectedValidationResults expectedResults)
         {
-            string[] detectedResultLocations = run.Results.Select(r => r.Message.Arguments[0]).OrderBy(loc => loc).ToArray();
-            string[] expectedResultLocations = expectedResults.ResultLocationPointers.OrderBy(loc => loc).ToArray();
+            HashSet<string> actualResultLocations = new HashSet<string>(run.Results.Select(r => r.Message.Arguments[0]).OrderBy(loc => loc));
+            HashSet<string> unexpectedlyAbsentResultLocations = new HashSet<string>(expectedResults.ResultLocationPointers.OrderBy(loc => loc));
 
-            // We could make this assertion at the start of the method. We delay it until here
-            // so that, during debugging, you can set a breakpoint here, and you'll have the
-            // detected and expected result location arrays available to compare.
-            run.Results.Count.Should().Be(expectedResults.ResultCount);
+            List<string> unexpectedNewIssues = new List<string>();
 
-            detectedResultLocations.Should().ContainInOrder(expectedResultLocations);
+            foreach(string location in actualResultLocations)
+            {
+                if (unexpectedlyAbsentResultLocations.Contains(location))
+                {
+                    // Happy path. Actual location found in expected
+                    unexpectedlyAbsentResultLocations.Remove(location);
+                }
+                else
+                {
+                    // Bad new unexpected issue 
+                    unexpectedNewIssues.Add(location);
+                }
+            }
+
+            // No new issues
+            unexpectedNewIssues.Count.Should().Be(0);
+
+            // No unexpectedly absent issues. If we found everything we expected,
+            // then our expected results should be empty
+            unexpectedlyAbsentResultLocations.Count.Should().Be(0);
```

Comment:
>unexpectedlyAbsentResultLocations [](start = 12, length = 33)

This whole thing can be done with a couple of LINQ calls. Also, a suggested variable rename for consistency:
```
IEnumerable<string> unexpectedNewResultLocations = actualResultLocations.Except(expectedResults.ResultLocationPointers);
unexpectedNewResultLocations.Count().Should().Be(0);

IEnumerable<string> unexpectedlyAbsentResultLocations = expectedResults.ResultLocationPointers.Except(actualResultLocations);
unexpectedlyAbsentResultLocations.Count().Should().Be(0)
```;

Reply:
I considered this but I don't like leaving O(N^2) comparisons hanging around. The code as authored is much less compact/elegant but is clear and performant. 

---
In reply to: [248362142](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248362142) [](ancestors = 248362142)

Reply:
Would you at least do the variable rename?

---
In reply to: [248384220](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248384220) [](ancestors = 248384220,248362142)